sunpy/instr/aia/response.py

```diff
@@ -0,0 +1,313 @@
+    filename to the temperature response function method.
+
+    References
+    ----------
+    .. [1] Boerner et al., 2012, Sol. Phys., `275, 41
+        <http://adsabs.harvard.edu/abs/2012SoPh..275...41B>`_
+    """
+
+    def __init__(self, channel_list=None, instrument_files=None):
+        if channel_list is None:
+            channel_list = [94, 131, 171, 193, 335, 211, 304]
+        self._get_channel_info(channel_list, instrument_files)
+
+    def _get_channel_info(self, channel_list, instrument_files):
+        """
+        Get instrument info for channels in channel list.
+
+        Download the instrument files if needed or if they were not specified at the command
+        line. Creates the dictionary `self._channel_info`.
+        """
+        if instrument_files is None:
+            instrument_files = []
+            download_dir = get_and_create_download_dir()
+            for instr_file in AIA_INSTR_FILES:
+                check_download_file(instr_file, SSW_AIA_REMOTE_PATH, download_dir=download_dir)
+                instrument_files.append(os.path.join(download_dir, instr_file))
+
+        data_table = aia_instr_properties_to_table(channel_list, instrument_files)
+        self._channel_info = {}
+        for c in channel_list:
+            index = channel_list.index(c)
+            self._channel_info[c] = {property: data_table[property][index]
+                                     for property in data_table.columns.keys()}
+
+    def calculate_effective_area(self, channel, include_crosstalk=True, **kwargs):
+        """
+        Calculate the effective area for a given channel of the instrument.
+
+        According to [1]_, the effective area is given by,
+
+        .. math::
+            A_{eff}(\lambda,t) = A_{geo}R_P(\lambda)R_S(\lambda)T_E(\lambda)
+            T_F(\lambda)D(\lambda,t)Q(\lambda),
+
+        in units of :math:`\mathrm{cm}^2`, where,
+
+        - :math:`A_{geo}`: geometrical collecting area
+        - :math:`R_P`, :math:`R_S`: reflectances of primary and secondary mirrors, respectively
+        - :math:`T_E`, :math:`T_F`: transmission efficiency of the entrance and focal-plane filters, respectively
+        - :math:`D`: correction to account for time-dependent contamination and deterioration
+        - :math:`Q`: quantum efficiency of the CCD
+
+        The effective area contains information about the efficiency of the
+        telescope optics and its sensitivity as a function of wavelength. All of the telescope
+        properties are read from the AIA instrument files available in SolarSoft.
+
+        Parameters
+        ----------
+        channel : `int`
+            wavelength center of the channel
+        include_crosstalk : `bool`
+
+        Returns
+        -------
+        effective_area : array-like
+            effective area of the instrument over the entire wavelength range
+        """
+        reflectance = (self._channel_info[channel]['primary_mirror_reflectance']
```

Comment:
I would create a variable as `mychannel = self._channel_info[channel]` to make this block more readable and lest repetition.